packages/node-resolve/src/index.js

```diff
@@ -267,9 +266,13 @@ export function nodeResolve(opts = {}) {
           importer,
           Object.assign({ skipSelf: true }, resolveOptions)
         );
-        const isExternal = !!(resolvedResolved && resolvedResolved.external);
-        if (isExternal) {
-          return false;
+        if (resolvedResolved) {
+          // Handle plugins that manually make the result external
+          if (resolvedResolved.external) {
+            return false;
+          }
+          // Pass on meta information added by other plugins
```

Comment:
I wonder if we should really just target property holding the metadata instead of allowing everything through by default except `moduleSideEffects`? Don't think we'd be doing any object merging then

Reply:
Admittedly, the last remaining property is `syntheticNamedExports`. Which we may want to keep IF we also keep the id. So maybe a better solution is: Keep everything from the second resolve, but override moduleSideEffects if the id did not change?